osquery/filesystem/fileops.h

```diff
@@ -72,6 +83,29 @@ typedef struct { PlatformTimeType times[2]; } PlatformTime;
 /// Constant for an invalid handle.
 const PlatformHandle kInvalidHandle = (PlatformHandle)-1;
 
+typedef struct win_stat {
+  std::string path;
+  std::string filename;
+  int symlink;
+  std::string file_id;
+  LONGLONG inode;
```

Comment:
This is what's breaking the posix builds, make these all `long long` instead of `LONGLONG`? Will that still build?

Reply:
Actually should it just be within an  #ifdef WIN32 block?  LONGLONG seems to be used elsewhere without issues unless I'm mistaken

Reply:
Maybe I don't understand the fileops abstraction completely. Should that struct and platformStat both be gated off for Windows only?

Reply:
Yeah definitely could be gated off or defined in the windows fileops abstraction code. I was just curious if you could change the preprocessor macros to get it compiling on Linux aswell, but it’d be better suited as you’ve described :)

Reply:
Ok so I made those changes and the POSIX builds pass, but not Windows. Is there a way so see what's causing the build failure? In my dev env I get these messages at the end of the build script
```
Ran 10 tests in 28.016s

FAILED (errors=2)

    Start 6: python_test_windows_service
6/6 Test #6: python_test_windows_service ......   Passed   22.31 sec

50% tests passed, 3 tests failed out of 6

Total Test time (real) = 441.75 sec

The following tests FAILED:
          1 - osquery_tests (Failed)
          2 - osquery_additional_tests (Failed)
          5 - python_test_osqueryd (Failed)
Errors while running CTest
```

Although the binaries seem to run just fine so I'm not entirely sure what's exactly failing.

Reply:
@Rich5 yeah you can click on the Windows build failures, it'll take you to the Jenkins build pages, from there you can click on the build, then on the `Console Output` link, and it'll show you the build results. I pulled down your branch and meant to check this out yesterday but didn't get a chance, and am looking into it now.